### PR TITLE
Add tests for non-implemented rmw_take_* functions

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

File: test_rmw_implementation/test/test_subscription.cpp [hunk 1]
```diff
@@ -377,3 +377,55 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
...
+  }
+
+  void TearDown() override
+  {
+    Base::TearDown();
+  }
+};
+
+TEST_F(CLASSNAME(TestSubscriptionUseLoan, RMW_IMPLEMENTATION), rmw_take_loaned_message) {
+  // TODO(lobotuerk) add tests for <rmw_take_loaned_message> when we have an implementation
```
Comment (hidmic) [inline test_rmw_implementation/test/test_subscription.cpp:419]: @Lobotuerk nit: 
```suggestion
  // TODO(lobotuerk): add tests for rmw_take_loaned_message() when we have an implementation.
```
Same elsewhere.

File: test_rmw_implementation/test/test_subscription.cpp [hunk 2]
```diff
@@ -377,3 +377,55 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
...
+
+  void SetUp() override
+  {
+    Base::SetUp();
+    // Check if loaning is supported by the implementation
+    if (!sub->can_loan_messages) {
+      bool taken = false;
+      void * loaned_message = nullptr;
+      rmw_message_info_t message_info = rmw_get_zero_initialized_message_info();
+      rmw_subscription_allocation_t * null_allocation{nullptr};
```
Comment (hidmic) [inline test_rmw_implementation/test/test_subscription.cpp:395]: @Lobotuerk nit: 
```suggestion
      rmw_subscription_allocation_t * null_allocation{nullptr};  // also valid
```

**Review by hidmic — APPROVED** [review]
LGTM but a for a few nits. Also, the PR title could be a bit more descriptive.

File: test_rmw_implementation/test/test_subscription.cpp (hunk 1 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp:419]: Updated

**Review by Blast545 — APPROVED** [review]
LGTM

**Comment by Lobotuerk** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12399/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7367/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10110/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12312/)

**Comment by Lobotuerk** [comment]
CI green, merging

**Pull request merged**